File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/sentinels/domain/SentinelsObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.sentinels.domain;


public abstract class SentinelsObject {
    public abstract String objAsString();
    public static SentinelsObject objFromString(String str) {
        switch (str) {
            case "marker":
                return new SentinelsMarkerObject();
            case "tower":
                return new SentinelsTowerObject();
            default:
                return new SentinelsMarkerObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/sentinels/SentinelsObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.sentinels

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class SentinelsObject {
    Empty, Forbidden, Hint, Marker, Tower
}

class SentinelsGameMove(val p: Position, var obj: SentinelsObject = SentinelsObject.Empty)
